File: backend/src/modules/academic/domain/assessment-period.ts
export type AssessmentPeriod = {
  id: string;
  teacherId: string;
  academicYearId: string;
  /** Extensão futura: override por turma (null = período do ano). */
  classId: string | null;
  name: string;
  sortOrder: number;
  startsOn: Date | null;
  endsOn: Date | null;
  createdAt: Date;
  updatedAt: Date;
  deletedAt: Date | null;
};

File: frontend/lib/domain/entities/assessment_period.dart
class AssessmentPeriod {
  const AssessmentPeriod({
    required this.id,
    required this.academicYearId,
    this.classId,
    required this.name,
    required this.sortOrder,
    this.startsOn,
    this.endsOn,
  });

  final String id;
  final String academicYearId;
  final String? classId;
  final String name;
  final int sortOrder;
  final DateTime? startsOn;
  final DateTime? endsOn;
}
